### PR TITLE
NimBLE AFR update and provision to use IRAM memory as 8-bit addressable memory region

### PR DESCRIPTION
**Pull request by sachin0x18**
Description
-----------
**Add Host based Privacy feature support:**

This feature removes dependency on controller to use privacy (RPA).
Add support to configure RPA Timeout from menuconfig (default is 15 minutes)
How to use this feature:

-  Introduced API to configure privacy : `ble_hs_pvcy_rpa_config()`
- `own_addr_type` needs to be set to BLE_ADDR_RANDOM while advertising/scanning.
Apart from 'Host based privacy', this PR has few miscellaneous feature and fixes.
- Configure Host (ble_hs_cfg) to distribute IRKs (BLE_SM_PAIR_KEY_DIST_ID)
Apart from 'Host based privacy', this PR has few miscellaneous feature and fixes.

**Use IRAM as Data memory**

`IRAM` in ESP32 is 32-bit addressable memory region and performing any unaligned (8-bit or 16-bit) load/store instructions on this region results in either `LoadStoreError` or `LoadStoreAlignment` exception. 
With the LoadStore exception handlers in place (inside `freertos-kernel`), IRAM can be used as 8-bit addressable memory region to store data. Performing unaligned access in IRAM will incur a maximum penalty of 167 cpu cycles.

- **Dynamic memory allocation:**
      -- Use dynamic buffers instead of static memory.
      -- Added IRAM allocation provision under `nimble_platform_mem_malloc()`.

 - **Allocate mbedTLS buffers in IRAM:**
      -- Allocate mbedTLS input and output buffers in the IRAM region thus saving ~12kB (with the default configuration) in DRAM.
    To enable this strategy, 

    `idf.py menuconfig -> Component config -> ESP32 Specific -> Enable IRAM as 8 bit accessible memory`

     ` idf.py menuconfig -> Component config -> mbedTLS -> Memory allocation strategy -> Internal IRAM`

Checklist:
----------
- [x] I have tested my changes. No regression in existing tests.

- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by sachin0x18** [comment]
This PR depends on https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/9 as well as  https://github.com/aws/amazon-freertos/pull/1709

File: .gitmodules [hunk 1]
```diff
@@ -8,8 +8,8 @@
 	branch = v2.4.3
 [submodule "freertos_kernel"]
 	path = freertos_kernel
-	url = https://github.com/FreeRTOS/FreeRTOS-Kernel.git
-	branch = V10.2.1-convergence-FreeRTOS-Source
+	url = https://github.com/sachin0x18/FreeRTOS-Kernel.git
```
Comment (lundinc2) [inline .gitmodules:11]: Before this gets merged this will need to be changed back

File: libraries/freertos_plus/standard/crypto/src/iot_crypto.c [hunk 2]
```diff
@@ -256,7 +256,7 @@ void CRYPTO_ConfigureHeap( void )
     /*
      * Ensure that the FreeRTOS heap is used.
      */
-    mbedtls_platform_set_calloc_free( prvCalloc, vPortFree ); /*lint !e534 This function always return 0. */
+    //mbedtls_platform_set_calloc_free( prvCalloc, vPortFree ); /*lint !e534 This function always return 0. */
```
Comment (lundinc2) [inline libraries/freertos_plus/standard/crypto/src/iot_crypto.c:259]: This change won't be necessary once this PR has been rebased to the master branch

**Comment by ravibhagavandas** [comment]
Verified the nimble changes by running the tests. To enable RPA feature , it also requires the fix mentioned in this PR: https://github.com/apache/mynewt-nimble/pull/790

**Comment by mahavirj** [comment]
@ravibhagavandas Requested change will be followup commit, so we can merge this PR interim. @prasad-alatkar Please confirm.

**Comment by prasad-alatkar** [comment]
@ravibhagavandas @mahavirj Yes, I am planning a new PR for this commit along with other changes (under progress).

**Comment by lundinc2** [comment]
Hello, 

Can you try re-basing your branch to the latest master branch? It seems some files unrelated to this PR are causing the style check to fail.

Thank you

**Comment by lundinc2** [comment]
Hello,

Apologies for the inconvenience. This PR fixes the styling issue https://github.com/aws/amazon-freertos/pull/1961. So one last rebase should do the trick. I will continue to investigate what has caused the issues to surface with this particular PR.

**Pull request merged**